packages/flutter_blue/tizen/src/bluetooth_device_controller.h

```diff
@@ -0,0 +1,103 @@
+#ifndef FLUTTER_BLUE_TIZEN_BLUETOOTH_DEVICE_CONTROLLER_H
+#define FLUTTER_BLUE_TIZEN_BLUETOOTH_DEVICE_CONTROLLER_H
+
+#include "GATT/bluetooth_service.h"
+#include "bluetooth.h"
+#include "flutterblue.pb.h"
+#include "notifications_handler.h"
+#include "utils.h"
+
+namespace flutter_blue_tizen {
+
+class BluetoothDeviceController {
+ public:
+  enum class State {
+    kConnected,
+    kConnecting,
+    kDisconnected,
+    kDisconnecting,
+  };
+
+  using requestMtuCallback =
+      std::function<void(bool, const BluetoothDeviceController&)>;
+
+  BluetoothDeviceController(
+      const std::string& address,
+      NotificationsHandler& notifications_handler) noexcept;
+
+  BluetoothDeviceController(
+      const char* address,
+      NotificationsHandler& notifications_handler) noexcept;
+
+  ~BluetoothDeviceController() noexcept;
+
+  BluetoothDeviceController() = delete;
+
+  BluetoothDeviceController(const BluetoothDeviceController& address) = delete;
```

Comment:
Any reason for deleting copy constructor?

Reply:
I don't follow, what's got to do with `std::vector<std::unique_ptr<btGatt::PrimaryService>> services_`?

Reply:
`BluetoothDeviceController`, has `std::unique_ptr<>` in a vector container. It makes it non copyable object and default copy ctor or `operator=` are implicitly deleted.